### DISCUSSION TITLE
Bug: action.yml runs.image is not used if Dockerfile is present

### DISCUSSION THREAD

**Discussion by judoole** [community/community#27099] [Actions]
<p>From the documentation of creating an action you can use a docker-image instead of a Dockerfile<a href="https://help.github.com/en/articles/metadata-syntax-for-github-actions#image" rel="nofollow noopener">https://help.github.com/en/articles/metadata-syntax-for-github-actions#image</a></p>
<p>This works fine, and greatly increases the speed of workflows. But when a Dockerfile is present in the repository, the Dockerfile is used instead. </p>
<p>The workaround seems to be to rename the Dockerfile.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/judoole">@judoole</a>,</p>
<p>Thank you for being here, and apologies for the delay in response. We appreciate you taking the time to report this, are you still having the same issue?</p>

**Comment by judoole** [comment]
<p>No stress <a class="mention" href="/andreagriffiths11">@andreagriffiths11</a> </p>
<p>The workaround does work, so there was no major worries. That thing has unfortunately been rewritten, so it’s a bit cumbersome to recreate for me again. More of a heads up to the dev-team or whomever else stumbles upon the same issue.</p>

**Comment by chrispat** [comment, marked as answer]
<p><a class="mention" href="/judoole">@judoole</a> Thank you for reporting this bug.  We have fixed and the updated runner should now be out to all production. </p>
<p>I apologize for the inconvenience. </p>

**Comment by michaelbeaumont** [comment]
<p><a class="mention" href="/chrispat">@chrispat</a>  This bug seems to have resurfaced:</p>
<pre><code>Run ./folder/migrate
Dockerfile for action: '/home/runner/work/repo/repo/./folder/migrate/./action.Dockerfile'.
/usr/bin/docker build -t 10865d:ab069054696f58c6f8e0f3a99e770579 "/home/runner/work/repo/repo/folder/migrate"
</code></pre>
<p>Which then fails because it actually uses the <code>Dockerfile</code> in <code>folder/migrate</code> and not <code>action.Dockerfile</code>. Seems that the <code>docker build</code> command is incorrect.</p>

**Comment by thboop** [comment]
<p><a class="mention" href="/michaelbeaumont">@michaelbeaumont</a> ,</p>
<p>The original issue was around the dockerfile being used when an image was specified if a dockerfile was present in the repo.</p>
<p>It sounds like you are seeing the wrong dockerfile is being used.</p>
<p>Could you provide a link to a run showcasing this issue or provide your action.yml <em>runs</em> section and the workflow showing how you invoke the action?</p>
<p>It looks like you are running a local action via </p>
<pre><code>uses: ./folder/migrate
</code></pre>

**Comment by michaelbeaumont** [comment]
<pre><code>name: "migrate"
runs:
  using: "docker"
  image: "./action.Dockerfile"

- uses: ./folder/migrate
        name: Handle migration
</code></pre>
<p>Indeed it’s a different situation but it seemed to me to be the same error, that <code>runs.image</code> is simply ignored.</p>
<p>In <code>folder/migrate</code> are <code>Dockerfile</code> and <code>action.Dockerfile</code></p>

**Comment by neilyio** [comment]
<p>This is still happening for me. If there’s a Dockerfile in the same folder as a action.yml, then the run.image key will be completely ignored.</p>
<p>In fact, as long as a Dockerfile is in the same folder, you can omit the runs.image key altogether. Depsite the documentation saying that runs.image is required.</p>
